### ISSUE TITLE
Can't avoid fp16 conversion of an onnx node

### ISSUE THREAD

**Issue by omera-deci** [NVIDIA/TensorRT#2899]
## Description

Hi!
I'm trying to use the python API to convert an onnx model (a BART encoder exported to onnx from pytorch) to trt with fp16/32 mixed-precision.
My model has a single weight that is out of the fp16 range, and so I wish to leave this weight as fp32 while converting the rest of the network to fp16. Reading the docs led me to the following code:
```python
#!/usr/bin/env python3
from contextlib import contextmanager
from pathlib import Path
import tensorrt as trt

EXPLICIT_BATCH = 1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH)
LOGGER = trt.Logger()
LOGGER.min_severity = trt.Logger.Severity.VERBOSE

LAYER_WITH_LARGE_WEIGHT = "/model/Constant_14_output_0"


@contextmanager
def trt_timing_cache(builder_config, cache_file_path):
    cache_path = Path(cache_file_path)
    cache_data = b""
    if cache_path.exists():
        with cache_path.open("rb") as cache_file:
            cache_data = cache_file.read()

    timing_cache = builder_config.create_timing_cache(cache_data)
    builder_config.set_timing_cache(timing_cache, ignore_mismatch=False)
    yield
    with cache_path.open("wb") as cache_file:
        cache_file.write(timing_cache.serialize())


def convert(onnx_path):
    with trt.Builder(LOGGER) as builder, builder.create_network(
        EXPLICIT_BATCH
    ) as network, builder.create_builder_config() as config, trt.OnnxParser(
        network, LOGGER
    ) as parser, trt_timing_cache(
        config, "timing_cache.dat"
    ):
        # parse onnx file
        with open(onnx_path, "rb") as onnx_model:
            parser_success = parser.parse(onnx_model.read())
        if not parser_success:
            raise RuntimeError(
                f"Failed parsing onnx file with the following errors: {[parser.get_error(i) for i in range(parser.num_errors)]}"
            )

        # set dynamic input shapes
        profile0 = builder.create_optimization_profile()
        for name in ["input_ids", "attention_mask"]:
            profile0.set_shape(name, min=(1, 1), opt=(1, 512), max=(1, 1024))
        config.add_optimization_profile(profile0)

        # leave the weight's precision as-is
        for layer in network:
            if layer.name == LAYER_WITH_LARGE_WEIGHT:
                layer.precision = layer.precision
                # also tried:
                # layer.precision = trt.float32
                # and also tried:
                # for i in layer.num_outputs:
                #     layer.set_output_dtype(i, layer.get_output(i).dtype)
                assert layer.precision_is_set
                break
        else:
            raise RuntimeError(f"Layer {LAYER_WITH_LARGE_WEIGHT} wasn't found in the network")

        # set model conversion to use fp16
        config.set_flag(trt.BuilderFlag.FP16)
        # also tried STRICT_SHAPES, but docs say it's deprecated
        # config.set_flag(trt.BuilderFlag.STRICT_TYPES)
        # these 3 flags mimic STRICT_SHAPES
        config.set_flag(trt.BuilderFlag.PREFER_PRECISION_CONSTRAINTS)
        config.set_flag(trt.BuilderFlag.DIRECT_IO)
        config.set_flag(trt.BuilderFlag.REJECT_EMPTY_ALGORITHMS)
        # also tried OBEY_PRECISION_CONSTRAINTS instead of PREFER

        config.set_preview_feature(trt.PreviewFeature.FASTER_DYNAMIC_SHAPES_0805, True)
        config.profiling_verbosity = trt.ProfilingVerbosity.DETAILED

        # convert to trt
        engine = builder.build_engine(network, config)

        return engine


if __name__ == "__main__":
    engine = convert("bart_encoder.onnx")
    with open("bart_encoder_fp16.engine", "wb") as f:
        f.write(bytes(engine.serialize()))

```

The layer name is the one specified in the onnx model. In the verbose output I can see that the layer was converted to fp16, even though I explicitly set its precision to fp32. I can also see that the layer name was changed (I guess due to it getting fused with another layer):
```
...
[04/24/2023-00:21:06] [TRT] [W] - 1 weights are affected by this issue: Detected finite FP32 values which would overflow in FP16 and converted them to the closest finite FP16 value.
[04/24/2023-00:21:06] [TRT] [V]   List of affected weights: /model/Constant_14_output_0 + (Unnamed Layer* 97) [Shuffle]
...
```

 Unfortunately, this weight overflow is hurting the converted model's accuracy (a.k.a it produces complete non-sense) and I am therefore unable to convert this model to fp16.

I also tried converting the model with polygraphy to no-avail:
```shell
polygraphy convert --convert-to trt --fp16 bart_encoder.onnx -o bart_encoder_fp16.engine --layer-precisions /model/Constant_14_output_0:float32 --precision-constraints prefer
```
How can I convert my model to fp16 without converting the problematic weight and without retraining?

## Environment

**TensorRT Version**: 8.5.3.1
**NVIDIA GPU**: NVIDIA RTX A4000
**NVIDIA Driver Version**: 515.43.04
**CUDA Version**: 11.7
**CUDNN Version**: 8.5.0
**Operating System**: Ubuntu 22.04
**Python Version (if applicable)**: 3.10
**Tensorflow Version (if applicable)**: N/A
**PyTorch Version (if applicable)**: 2.0.0
**Baremetal or Container (if so, version)**: Baremetal

## Relevant Files

[verbose_conversion_logs.txt](https://github.com/NVIDIA/TensorRT/files/11304982/verbose_conversion_logs.txt)

## Steps To Reproduce
1. Download the onnx model from https://drive.google.com/drive/folders/1B9X6KH6lObLt1a-udtAueXZjhfIFSBsp?usp=sharing and save it as "bart_encoder.onnx".
2. Run the script in the description in the same directory as the downloaded model.

**Comment by zerollzeng** [comment]
the subnormal weight comes from here:
![image](https://user-images.githubusercontent.com/38289304/234596870-b9c49109-fcbc-4add-8e3a-9815302dc14d.png)

**Comment by zerollzeng** [comment]
@pranavm-nvidia I tried `polygraphy convert --convert-to trt --fp16 bart_encoder.onnx -o bart_encoder_fp16.engine --layer-precisions /model/Constant_14_output_0:float32 /model/Where_1:float32 --precision-constraints obey` but it still throw the warning, do you know why? Thanks!

**Comment by pranavm-nvidia** [comment]
@zerollzeng This value would propagate to subsequent layers, so you would have to run all of those in FP32 as well. Although I'm not sure this is really the cause of the accuracy degradation. It seems like this value is being used like a `-inf` so I don't know if it would actually be involved in any computation later on. What kind of ops does this `Where` feed into?

**Comment by omera-deci** [comment]
While it would be nice to know how to prevent specific layers from being convefrtd, @pranavm-nvidia is right and it seems the accuracy degradation is not caused by this weight (opened https://github.com/NVIDIA/TensorRT/issues/2922 for that).

**Comment by zerollzeng** [comment]
Can we close this issue? Thanks!

**Comment by omera-deci** [comment]
Yes, I'll continue in https://github.com/NVIDIA/TensorRT/issues/2922 . Thank you :)
